src/jdk.jlink/share/classes/jdk/tools/jlink/internal/plugins/SystemModulesPlugin.java

```diff
@@ -1058,13 +1011,14 @@ void build() {
             }
 
             void newBuilder() {
-                mv.visitTypeInsn(NEW, MODULE_DESCRIPTOR_BUILDER);
-                mv.visitInsn(DUP);
-                mv.visitLdcInsn(md.name());
-                mv.visitMethodInsn(INVOKESPECIAL, MODULE_DESCRIPTOR_BUILDER,
-                    "<init>", "(Ljava/lang/String;)V", false);
-                mv.visitVarInsn(ASTORE, BUILDER_VAR);
-                mv.visitVarInsn(ALOAD, BUILDER_VAR);
+                cob.new_(CD_MODULE_BUILDER)
+                   .dup()
+                   .constantInstruction(md.name())
+                   .invokespecial(CD_MODULE_BUILDER,
+                                  "<init>",
+                                  MethodTypeDesc.of(CD_void, CD_String))
+                   .astore(BUILDER_VAR)
+                   .aload(BUILDER_VAR);
```

Comment:
The methods to generate the MD all load the Builder onto the operand stack so the aload at L1021 is catching my attention, is that needed?

Reply:
It seems to be redundant as the following code does not expect it on stack and loads it again as needed.

Reply:
I can confirm `java -XX:+UnlockDiagnosticVMOptions -XX:+BytecodeVerificationLocal -version` successful execution and passing all  jdk/tools/jlink tests. Now waiting for all tier1 tests results.

Reply:
> Now waiting for all tier1 tests results.

The jlink tests run in tier2 so maybe sure to run those.

Reply:
all tier1, 2, and 3 tests passed